### ISSUE TITLE
Just for clarification

### ISSUE THREAD

**Issue by TimoKramer** [ioxiocom/openapi-to-fastapi#22]
Hi,
I looked into your code. This project seems to be what I was searching for. I see that it is not for general audience.

Just for my clarification: Is this project a validator of what (POST-)routes I am writing? I was searching for a project that basically saves me defining my routes by hand, so I only need to define the corresponding handlers (maybe connecting them based on the operationID).

Is it (only) a validator?

Thanks for this project, it helps even if it is (only) a validator.

**Comment by fbjorn** [comment]
Hi @TimoKramer. 
I would say that validation of OpenAPI specs is a side effect of this project (however we also created a CLI helper for such validations and use it as part of our CI).
Yes, you're right, this is not for general audience cause it solves pretty specific need: if you have OpenAPI spec(s) and you want to create an application based on these spec(s). Usually you write the code first, and then generate swagger schemas as an output :-) the situation is reversed there. If you have a specific need - for example creating a proxy to 3rd party services using 3rd party specs, this tool can be useful. Otherwise I doubt it worth the effort.

Not sure if I understood you correctly, but yes, if you have an OpenAPI spec, then this tool can create all the routes for you automatically, and you can implement required handlers. Check this section in the readme: https://github.com/ioxiocom/openapi-to-fastapi#custom-routes

If you have other questions - feel free to ask, I will be happy to help

**Comment by TimoKramer** [comment]
That's actually pretty cool. Thanks! I've misunderstood the 'Custom routes' heading and didn't read it. It's pretty much what I need.
I like to specify the contract first in openapi and with this lib I can take the spec and just create the handlers. That's amazing...

Is there something specific that keeps you from implementing other http request methods? Just too much work and you don't need it?

**Comment by fbjorn** [comment]
Yeah, we only needed POST routes for our project. To be honest, I didn't think this repo will gain some popularity (right now it has 56 stars which is already something!). You can submit a feature request for adding other types of routes and we will think when to allocate time for that 🙂
